src/app/modules/ix-table2/array-data-provider.ts

```diff
@@ -0,0 +1,72 @@
+import * as _ from 'lodash';
+import { Observable, map, of } from 'rxjs';
+import { TablePagination } from 'app/modules/ix-table2/interfaces/table-pagination.interface';
+import { TableProvider } from 'app/modules/ix-table2/interfaces/table-provider.interface';
+import { TableSort } from 'app/modules/ix-table2/interfaces/table-sort.interface';
+
+export class ArrayDataProvider<T> implements TableProvider<T> {
+  currentPage$: Observable<T[]>;
+  rows$: Observable<T[]> = of([] as T[]);
+  expandedRow: T;
+
+  sorting: TableSort<T> = {
+    propertyName: null,
+    direction: null,
+    active: null,
+  };
+
+  pagination: TablePagination = {
+    pageNumber: null,
+    pageSize: null,
+  };
+
+  setRows(rows$: Observable<T[]>): void {
+    this.rows$ = rows$;
```

Comment:
I feel like ArrayDataProvider shouldn't be accepting data as observables and should just accept static values.
In the future we may consider having something like `AsyncDataProvider` that would accept `this.ws.call()` directly.